### DISCUSSION TITLE
mysql connection via api route

### DISCUSSION THREAD

**Discussion by adityafawzan** [nuxt/nuxt#17986] [Questions] [answered]
is it possible to connect mysql via api route ? im getting error, [proxy] socket hang up

import mysql from "mysql";
export default (req, res) => {
  var con = mysql.createConnection({
    host: "localhost",
    user: "user",
    password: "password",
    database: "db",
  });

  con.connect(function (err) {
    if (err) throw err;
    console.log("Connected!");
  });
};

**Comment by wobsoriano** [comment, marked as answer]
Well, first of all, you're not returning any data in your api that's why you're getting a socket hang up. Second, please install the maintained version of [mysql](https://github.com/sidorares/node-mysql2) instead. 

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt3';

export default defineNuxtConfig({
  privateRuntimeConfig: {
    mysql: {
      host: process.env.MYSQL_HOST,
      user: process.env.MYSQL_USER,
      password: process.env.MYSQL_PASSWORD,
      database: process.env.MYSQL_DATABASE,
    },
  },
});
```

```ts
// server/api/products.ts
import mysql from 'mysql2/promise';
import config from '#config';

export default async (req, res) => {
  const con = await mysql.createConnection(config.mysql);
  const [rows, fields] = await con.execute('SELECT * FROM products');
  return {
    rows
  };
};

```

**Reply by wobsoriano** [reply]
You might want to check https://www.prisma.io/ btw

**Reply by folamy** [reply]
@wobsoriano, can you please help me with this nuxt/nuxt#17953

**Reply by adityafawzan** [reply]
big thanks @wobsoriano , u helped a lot

**Reply by adityafawzan** [reply]
one more thing, how to get the rows as JSON object in template @wobsoriano ?

<template>
  <div>
    <table>
      <thead>
        <tr>
          <th></th>
        </tr>
      </thead>
      <tbody>
        
      </tbody>
    </table>
  </div>
</template>

<script setup lang="ts">
const posts = await useFetch("/api/mysql");

console.log(posts);
</script>


it returns on console : 

{
  data: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: { result: [Array] },
    _value: { result: [Array] }
  },
  pending: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: false,
    _value: false
  },
  error: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: null,
    _value: null
  },
  refresh: [Function (anonymous)]
}

**Reply by adityafawzan** [reply]
@wobsoriano well after few tries, i managed to retrieve the result into template, but i dont know whether it's a best practice or not

<template>
  <div>
    <table>
      <thead>
        <tr>
          <th></th>
        </tr>
      </thead>
      <tbody>
        <tr v-for="post in posts" :key="post.id">
          <td>{{ post.kode }}</td>
        </tr>
      </tbody>
    </table>
  </div>
</template>

<script setup lang="ts">
const res = await useFetch("/api/mysql");
const posts = res.data.value.rows;
</script>

**Reply by wobsoriano** [reply]
That's right
